ultraplot/tests/test_colorbar.py

```diff
@@ -353,3 +354,135 @@ def test_label_placement_fig_colorbar2():
+    title = "My Label"
+    fig, ax = uplt.subplots()
+
+    cbar = ax.colorbar(cmap, loc=cbarloc, labelloc=labelloc, title=title)
+
+    x_label = cbar.ax.xaxis.label.get_text()
+    y_label = cbar.ax.yaxis.label.get_text()
+
+    assert title in (x_label, y_label), (
+        f"Expected label '{title}' not found. "
+        f"xaxis label: '{x_label}', yaxis label: '{y_label}', "
+        f"labelloc='{labelloc}', cbarloc='{cbarloc}'"
+    )
+
+    uplt.close(fig)
+
+
+@pytest.mark.parametrize(
+    ("cbarloc", "invalid_labelloc"),
+    product(
+        ["top", "bottom", "upper left", "lower right"],
+        ["invalid", "diagonal", "center", "middle", 123, "unknown"],
+    ),
+)
+def test_colorbar_invalid_horizontal_label(cbarloc, invalid_labelloc):
+    """
+    Test error conditions and edge cases for colorbar label placement.
+    """
+    cmap = uplt.Colormap("plasma_r")
+    title = "Test Label"
+    fig, ax = uplt.subplots()
+
+    # Test ValueError cases - invalid labelloc for different colorbar locations
+
+    # Horizontal colorbar location with invalid labelloc
+    with pytest.raises(ValueError, match="Could not determine position"):
+        ax.colorbar(cmap, loc=cbarloc, labelloc=invalid_labelloc, label=title)
+    uplt.close(fig)
+
+
+@pytest.mark.parametrize(
+    ("cbarloc", "invalid_labelloc"),
+    product(
+        ["left", "right", "ll", "ul", "ur", "lr"],
+        [
+            "invalid",
+            "diagonal",
+            "center",
+            "middle",
+            123,
+            "unknown",
+        ],
+    ),
+)
+def test_colorbar_invalid_vertical_label(cbarloc, invalid_labelloc):
+    # Vertical colorbar location with invalid labelloc
+    cmap = uplt.Colormap("plasma_r")
+    title = "Test Label"
+    fig, ax = uplt.subplots()
+    with pytest.raises(ValueError, match="Could not determine position"):
+        ax.colorbar(cmap, loc=cbarloc, labelloc=invalid_labelloc, label=title)
+    uplt.close(fig)
+
+
+@pytest.mark.parametrize(
+    "invalid_labelloc", ["fill", "unknown", "custom", "weird_location", 123]
+)
+def test_colorbar_invalid_fill_label_placement(invalid_labelloc):
+    # Fill location with invalid labelloc
+    cmap = uplt.Colormap("plasma_r")
+    title = "Test Label"
+    fig, ax = uplt.subplots()
+    with pytest.raises(ValueError, match="Could not determine position"):
+        ax.colorbar(cmap, loc="fill", labelloc=invalid_labelloc, label=title)
+
+
+@pytest.mark.parametrize("unknown_loc", ["unknown", "custom", "weird_location", 123])
+def test_colorbar_label_placement_default(unknown_loc):
+    # Test the default case (final else statement)
+    # Unknown/unrecognized locations should default to obj.set_label(label)
+    cmap = uplt.Colormap("plasma_r")
+    title = "Test Label"
+    fig, ax = uplt.subplots()
+    # This should not raise an error and should use the default behavior
+    with pytest.raises(KeyError):
+        cbar = ax.colorbar(cmap, loc=unknown_loc, label=title)
```

Comment:
The comment here doesn't match the test result which only passes if an error is raised.

Reply:
Marked the test wrong. Label is corrected.